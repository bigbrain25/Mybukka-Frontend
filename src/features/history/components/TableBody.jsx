import React from 'react';
import PropTypes from 'prop-types';
import Courier from '../common/Courier';
import PickUp from '../common/PickUp';
import Price from '../common/Price';
import Order from '../common/Order';
import Status from '../common/Status';
import DropOff from '../common/DropOff';

const Col = ({ children }) => (<div
  className="col-xl-2 col-lg-2 col-md-10 pt-3 border"
>
  {children}
</div>);

const Row = ({ children, handleClick }) => (
  <div
    onClick={handleClick}
    role="button"
    aria-pressed="false"
    tabIndex="0"
    className="row flex flex-xl-row flex-lg-row d-xl-flex d-lg-flex d-md-none
    d-none cursor-pointer"
  >
    {children}
  </div>
);

const TableBody = ({
  handleClick,
  time,
  status,
  quantity,
  title,
  price,
  orderId,
  courierName,
  courierImg,
  pickupAddress,
  pickupContactMobile,
  deliveryAddress,
  deliveryName,
  deliveryContactMobile
}) => (
  <Row
    handleClick={handleClick}
  >
    <Col>
      <Status status={status} time={time} />
    </Col>
    <Col>
      <PickUp
        title={title}
        pickupAddress={pickupAddress}
        pickupContactMobile={pickupContactMobile}
      />
    </Col>
    <Col>
      <DropOff
        deliveryName={deliveryName}
        deliveryContactMobile={deliveryContactMobile}
        deliveryAddress={deliveryAddress}
      />
    </Col>
    <Col>
      <Order quantity={quantity} orderId={orderId} />
    </Col>
    <Col>
      <Price price={price} />
    </Col>
    <Col>
      <Courier courierName={courierName} courierImg={courierImg} />
    </Col>
  </Row>
);

export default TableBody;

TableBody.propTypes = {
  handleClick: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  quantity: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  orderId: PropTypes.string.isRequired,
  courierName: PropTypes.string.isRequired,
  courierImg: PropTypes.string.isRequired,
  pickupAddress: PropTypes.string.isRequired,
  pickupContactMobile: PropTypes.string.isRequired,
  deliveryAddress: PropTypes.string.isRequired,
  deliveryName: PropTypes.string.isRequired,
  deliveryContactMobile: PropTypes.string.isRequired,
};

Col.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

Row.propTypes = {
  handleClick: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};
