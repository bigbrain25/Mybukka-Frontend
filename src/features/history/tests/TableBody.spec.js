import React from 'react';
import TableBody from '../components/TableBody';

describe('TableBody component', () => {
  const props = {
    quantity: '2',
    title: 'salad and bbq',
    handleClick: jest.fn(),
    time: '27/3/2019 12:43 pm',
    orderId: '03d77',
    price: '3000',
    status: 'pending',
    courierName: 'curier name',
    courierImg: 'https;//www.google.com/img',
    pickupAddress: 'pickup address street',
    pickupContactMobile: '03847858',
    deliveryAddress: 'delivery address street',
    deliveryName: 'mr delivery name',
    deliveryContactMobile: '94574783930',
  };

  const wrapper = mount(<TableBody {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
