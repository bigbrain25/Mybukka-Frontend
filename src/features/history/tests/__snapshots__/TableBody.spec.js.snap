// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody component renders properly 1`] = `ReactWrapper {}`;
